### ISSUE TITLE
(Naomi) MAME zip+chd games crash on Nvidia Shield due to ANR

### ISSUE THREAD

**Issue by ds22x** [libretro/flycast#904]
As the title suggest, MAME zip+chd games crash when trying to load them on Nvidia Shield, while zip only roms will load just fine.
Have tried the exact same games, settings and bios files on both an OnePlus 6 and a laptop with an Intel Celeron processor, and the zip+chd games will load just fine as well, albeit after several seconds have passed.

**Comment by flyinghead** [comment]
Similar to #712

**Comment by barbudreadmon** [comment]
Isn't that just the infamous android issue where some devices arbitrarily close apps if they don't render something new for too long ?

**Comment by flyinghead** [comment]
Good point. That may very well be the case here.

If it's android closing the app, retrying a couple of times will generally succeed due to data being loaded in the disk cache. Also this will only happen on large CHDs. Small ones should load without issue.

**Comment by ds22x** [comment]
I see.
I would like to add that it doesn't force close on the Shield when using the debug standalone Flycast apks.

**Comment by flyinghead** [comment]
Standalone flycast loads the zip/chd in a background thread and keeps updating the UI so android would have no reason to force close it during load.

What's the size of the CHD you are using? Have you tried more than one?

**Comment by ds22x** [comment]
The sizes of the chd files I've tried are 60mb, 78mb, and 119mb.

**Comment by flyinghead** [comment]
This is pretty small so I doubt RA is unresponsive for very long, especially on Nvidia Shield.

I'll try to reproduce the crash on my Shield when I get back. What android version are you using? Which Retroarch version?

**Comment by ds22x** [comment]
Android 9 (SHIELD Experience 8.0.2), and RetroArch 1.8.8.
And the games I tried were Monkey Ball, Virtua Tennis 2, and Cleopatra's Fortune Plus, all from the most recent MAME set I could find for them (0.218).

EDIT: It seems to be indeed an issue where the Shield force closes RetroArch before it can fully load the game, as enabling the "ANR's in the background" option inside developer options will prompt an ANR dialog box, which allows you to actually load the game when you select "wait".

**Comment by flyinghead** [comment]
Yes, I have the same experience on my Shield as well: RA is closed because of being ANR.

**Comment by flyinghead** [comment]
This commit may help: https://github.com/libretro/flycast/pull/906/commits/75bfd3247e438e6fe2aefbd35f8ff43ecdc7f161

**Comment by ds22x** [comment]
Just build it from source and tried it myself, but the ANR problem still persists.
EDIT: Just tried it using a different controller than the one I was previously using (going from a Dualshock 4 V2 connected via USB, to an 8bitdo sn30 pro+ connected via bluetooth), and the games load without any ANR occuring.
